sw/device/lib/crypto/impl/hash.c

```diff
@@ -211,127 +212,91 @@ crypto_status_t otcrypto_hash(crypto_const_byte_buf_t input_message,
     return OTCRYPTO_BAD_ARGS;
   }
 
-  // Check `digest->len` is consistent with `hash_mode`
-  HARDENED_TRY(check_digest_len(hash_mode, digest->len));
+  // Check that digest length and mode match.
+  HARDENED_TRY(check_digest_len(digest));
 
-  switch (hash_mode) {
+  switch (digest->mode) {
     case kHashModeSha3_224:
-      HARDENED_TRY(
-          kmac_sha3_224(input_message.data, input_message.len, digest->data));
-      break;
+      return kmac_sha3_224(input_message.data, input_message.len, digest->data);
     case kHashModeSha3_256:
-      HARDENED_TRY(
-          kmac_sha3_256(input_message.data, input_message.len, digest->data));
-      break;
+      return kmac_sha3_256(input_message.data, input_message.len, digest->data);
     case kHashModeSha3_384:
```

Comment:
Is there an implication/side-effect to removing these `HARDENED_TRY` macros? Does that mean that the caller of `otcrypto_hash` should take care of this?

Reply:
Not really. The caller is expected to compare the return value with the hardened "OK" status in some safe manner according to their threat model. If they're not doing that and the attacker can glitch their check, then it doesn't help to have an extra `HARDENED_TRY` here for the return value; the attacker can just glitch the caller's check later on.

Reply:
That sounds fine. I was not thinking about the threat scenario (faulting or glitching) at all, so my question was quite shallow.

I was just thinking if we can arbitrarily replace `HARDENED_TRY` with `return` statements. Then after a second look, I realized there there is nothing left to do in the function after the switch-case statement, so maybe that is why we do not need the hardened return check anymore.

Reply:
Yep, that's exactly it!